LibC/stdlib.cpp

```diff
@@ -46,67 +46,64 @@ void abort()
 
 char* getenv(const char* name)
 {
+    size_t vl = strlen(name);
     for (size_t i = 0; environ[i]; ++i) {
         const char* decl = environ[i];
         char* eq = strchr(decl, '=');
         if (!eq)
             continue;
         size_t varLength = eq - decl;
-        char* var = (char*)alloca(varLength + 1);
-        memcpy(var, decl, varLength);
-        var[varLength] = '\0';
-        if (!strcmp(var, name)) {
-            char* value = eq + 1;
-            return value;
+        if (vl != varLength)
+            continue;
+        if (strncmp(decl, name, varLength) == 0) {
+            return eq + 1;
         }
     }
     return nullptr;
 }
 
 int putenv(char* new_var)
 {
-    HashMap<String, String> environment;
+    char* new_eq = strchr(new_var, '=');
+
+    // FIXME: should remove the var from the environment.
```

Comment:
I can fix this now, or I can do it on top when adding unsetenv. What do you prefer?

Reply:
No preference, maybe a separate commit for unsetenv() idk?